src/store/selectors/deck-filters.ts

```diff
@@ -0,0 +1,277 @@
+  state.deckFilters.filters.faction;
+
+const filterDeckByFaction = (faction: string) => {
+  return (deck: ResolvedDeck) => {
+    return deck.cards.investigator.card.faction_code === faction;
+  };
+};
+
+const makeDeckFactionFilter = (values: MultiselectFilter) => {
+  return or(values.map((value) => filterDeckByFaction(value)));
+};
+
+// Tag
+const filterDeckByTag = (tag: string) => {
+  return (deck: ResolvedDeck) => {
+    return deck.tags.includes(tag);
+  };
+};
+
+const makeDeckTagsFilter = (values: MultiselectFilter) => {
+  return or(values.map((value) => filterDeckByTag(value)));
+};
+
+export const selectTagsChanges = createSelector(
+  selectDeckFilters,
+  (filters) => {
+    const tagsFilters = filters.tags;
+    if (!tagsFilters.length) return "";
+    return tagsFilters.map(capitalize).join(" or ");
+  },
+);
+
+// Properties
+export const selectDeckPropertiesFilter = (state: StoreState) =>
+  state.deckFilters.filters.properties;
+
+export const selectPropertiesChanges = createSelector(
+  selectDeckPropertiesFilter,
+  (properties) => {
+    return Object.keys(properties)
+      .filter((prop) => properties[prop as DeckPropertyName])
+      .map(capitalize)
+      .join(" and ");
+  },
+);
+
+const makeDeckPropertiesFilter = (
+  properties: Record<DeckPropertyName, boolean>,
+) => {
+  const filters = [];
+  for (const property of Object.keys(properties)) {
+    if (properties[property as DeckPropertyName]) {
+      switch (property) {
+        case "parallel":
+          filters.push((deck: ResolvedDeck) =>
+            Boolean(
+              deck.investigatorFront.card.parallel ||
+                deck.investigatorBack.card.parallel,
+            ),
+          );
+      }
+    }
+  }
+  return and(filters);
+};
+
+// Validity
+const makeDeckValidityFilter = (value: Omit<DeckValidity, "all">) => {
+  switch (value) {
+    case "valid":
+      return (deck: ResolvedDeck) => deck.problem === null;
+    case "invalid":
+      return (deck: ResolvedDeck) => Boolean(deck.problem);
+    default:
+      return (_: ResolvedDeck) => true;
+  }
+};
+
+// Exp Cost
+export const selectDecksMinMaxExpCost = createSelector(
+  selectLocalDecks,
+  (decks) => {
+    const minmax: RangeMinMax = decks.reduce<[number, number]>(
+      (acc, val) => {
+        const { xpRequired } = val.stats;
+        acc[0] = Math.min(acc[0], xpRequired);
+        acc[1] = Math.max(acc[1], xpRequired);
+        return acc;
+      },
+      [Number.POSITIVE_INFINITY, Number.NEGATIVE_INFINITY],
+    );
+    return minmax;
+  },
+);
+
+export const selectExpCostChanges = createSelector(
+  selectDeckFilters,
+  (filters) => {
+    const expMinMax = filters.expCost;
+    return expMinMax ? `${expMinMax[0]}-${expMinMax[1]} experience` : "";
+  },
+);
+
+export const makeDeckExpCostFilter = (minmax: [number, number]) => {
+  return (deck: ResolvedDeck) => {
+    return (
+      deck.stats.xpRequired >= minmax[0] && deck.stats.xpRequired <= minmax[1]
```

Comment:
There is a slight distinction between xpRequired and xpSpent here since there can be discounts. I think we can ignore this for now.